core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -508,13 +517,40 @@ void updateFee() throws SepValidationException, AnchorException {
                     .receiveAmount(null)
                     .senderId(request.getSenderId())
                     .receiverId(request.getReceiverId())
-                    .clientId(token.getAccount())
+                    .clientId(getClientName())
                     .build())
             .getFee();
     infoF("Fee for request ({}) is ({})", request, fee);
     Context.get().setFee(fee);
   }
 
+  String getClientName() throws BadRequestException {
+    return getClientName(Context.get().getSep10Jwt().getAccount());
+  }
+
+  String getClientName(String account) throws BadRequestException {
```

Comment:
Is `Sep31Service` the best place for this class?

Reply:
This function is to find the name of a client (whose domain is listed in SEP-10 config) through clientsConfig (listed in SEP-31). So it has to be here, or we can move it to a helper class